### PR TITLE
Update epoch setup random source

### PR DESCRIPTION
**Pull request by tarakby**
- fix a misalignment between the node software and the epoch core-contract: the core-contract emits an event containing the random source of the new epoch. The event includes the random source as a decimal string, while the node software expects a hexadecimal string. This does not cause a functional bug because every decimal string is a valid hexadecimal string.  However this reduces the entropy of the random source read in the node software. 
- increase the epoch random source entropy from 64 to 128 bits, to address a security improvement in https://github.com/dapperlabs/flow-go/issues/6150 (the entropy generated by the Flow random beacon is about 128 bits). Therefore the `randomSource` field in the `EpochStart` event is updated from a string of variable number of characters (up to 20) to a string of exactly 32 characters.

Side changes:
- use Cadence's `revertibleRandom` instead of `unsafeRandom`
- update Cadence, flow-go, flow-emulator versions in lib/go/test

### CODE REVIEW

**Comment by tarakby** [comment]
Sounds good. Feel free to block the PR and manage when you'd like to merge it.

File: lib/go/test/go.mod [hunk 1]
```diff
@@ -4,91 +4,111 @@ go 1.18
...
-	github.com/onflow/cadence v0.39.12
-	github.com/onflow/flow-core-contracts/lib/go/contracts v1.2.4-0.20230703193002-53362441b57d
-	github.com/onflow/flow-core-contracts/lib/go/templates v1.2.3
-	github.com/onflow/flow-emulator v0.51.2-0.20230704183611-ecad54e231b7
-	github.com/onflow/flow-go v0.31.1-0.20230704154018-87a84e9d36c2
-	github.com/onflow/flow-go-sdk v0.41.6
-	github.com/onflow/flow-go/crypto v0.24.7
+	github.com/onflow/cadence v0.42.5
+	github.com/onflow/flow-core-contracts/lib/go/contracts v1.2.4-0.20231016154253-a00dbf7c061f
+	github.com/onflow/flow-core-contracts/lib/go/templates v1.2.4-0.20231016154253-a00dbf7c061f
```
Comment (jordanschalm) [inline lib/go/test/go.mod:9]:
```suggestion
	// replaced by module version in this repo - disregard pinned version
	github.com/onflow/flow-core-contracts/lib/go/contracts v1.2.4-0.20231016154253-a00dbf7c061f
	// replaced by module version in this repo - disregard pinned version
	github.com/onflow/flow-core-contracts/lib/go/templates v1.2.4-0.20231016154253-a00dbf7c061f
```

We could put these two in a separate `require` block next to the `replace` block at the bottom to make this fact more obvious.

**Review by jordanschalm — APPROVED** [review]

File: lib/go/test/go.mod (hunk 1 above)
Comment (tarakby) [reply lib/go/test/go.mod]: https://github.com/onflow/flow-core-contracts/pull/397/commits/13605a229b143d3d715cc2bd276f4a518df261e5

Btw, the imported version of `flow-core-contract` was updated automatically after updating the imported `flow-go` version (which itself imports `flow-core-contract`)

File: contracts/epochs/FlowEpoch.cdc [hunk 2]
```diff
@@ -511,7 +511,13 @@ pub contract FlowEpoch {
 
             let proposedNodeIDs = FlowEpoch.endStakingAuction()
 
-            FlowEpoch.startEpochSetup(proposedNodeIDs: proposedNodeIDs, randomSource: unsafeRandom().toString())
+            /// random source must be a hex string of 32 characters (i.e 16 bytes or 128 bits)
+            /// `revertibleRandom` returns a UInt64 (8 bytes)
+            let randomLow = revertibleRandom().toBigEndianBytes()
```
[the remark this answers is not in the document]
Comment (jordanschalm) [reply contracts/epochs/FlowEpoch.cdc:516]: I think it is worthwhile to add an assert statement here as well. I'm generally in favour of being overly cautious when making sure we produce valid service events.

[the remark this answers is not in the document]
Comment (tarakby) [reply contracts/epochs/FlowEpoch.cdc:516]: https://github.com/onflow/flow-core-contracts/pull/397/commits/8703b9e4079ee646d5577a47ac7b90ca63d1a6e3

**Pull request merged**